template/9-design.md

```diff
@@ -2,23 +2,34 @@
 
 ## Frontend
 
-*List the key libraries, languages, components used by the MVP.*
+**Implementation Framework** The application has been developed with Flutter, which is based on the Dart programming language. Flutter can natively compile on Android and iOS from the same code base, which allows us to launch the application to more users easily. 
+We use the Model View ViewModel (MVVM) architecture to ensure structured management and interactions between the UI and the application logic. We use the Riverpod plugin, which simplifies the application of the Observer pattern, reducing necessary UI refreshes.
 
-- Dart
-- Flutter
-- Firebase
-- GeoFlutterFirePlus
+**Firebase communication** We use Firebase Authentication to handle user authentication. The Firebase Core, Firebase Auth and Google Sign-In Flutter plugins control the user authentication flow through a Google Sign-In intent on the Login screen.
 
-*If applicable, describe essential screens.*
+Storing and fetching the user interactions in the application are handled using the Firebase Core and Cloud Firestone Flutter plugins.
 
-- Login
-- Home
-- Profile
-- Post Creation
-- Post Details
-- Challenges
-- Map
+**User location processing** In order to retrieve the user's surrounding posts and display the map with the current user location, we use the GeoLocator Flutter plugin. This plugin simplifies access to the user's current location on every platform. Additionally, we use GeoFlutterFire Plus, which simplifies the storage and fetching of Firebase documents based on their location. 
 
+### Essential screens
+
+**Login page**: First screen that appears when the user is not authenticated. It allows the user to authenticate and access the application.
+
+**Feed**: The feed is the central screen of the application. Shown when starting the application as an authenticated user, it displays the posts that are at less than one hundred meters away from the user. This screen allows him to sort the feed by hottest, top, latest, or even nearest posts. The feed displays, for every post, its title, the first seven lines of its description, its owner, the number of comments, the date of the post along with its upvote score. It also allows the user to upvote or downvote any posts directly from the feed screen.
+
+**Profile page**: The profile screens displays all the information associated to the user, such as his username, display name, number of centauri points, badges, number of followers along with his number of followed users. Moreover, it displays the title and description of all of his posts along with all of his comments. The user is able to delete any post or comment that he created, from anywhere, directly from the profile page.
```

Comment:
"profile screen~s~"

Reply:
Thank you !

Fixed in: [615314c](https://github.com/ProximaEPFL/prd/pull/8/commits/615314c78843abeb9be39f8874b637ee9a70005a)